homeassistant/components/wilight/fan.py

```diff
@@ -0,0 +1,111 @@
+"""Support for WiLight Fan."""
+
+from pywilight.const import DIRECTION_OFF, DOMAIN, FAN_V1, ITEM_FAN
+
+from homeassistant.components.fan import (
+    DIRECTION_FORWARD,
+    SPEED_HIGH,
+    SPEED_LOW,
+    SPEED_MEDIUM,
+    SUPPORT_DIRECTION,
+    SUPPORT_SET_SPEED,
+    FanEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from . import WiLightDevice
+
+SUPPORTED_SPEEDS = [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
+
+SUPPORTED_FEATURES = SUPPORT_SET_SPEED + SUPPORT_DIRECTION
+
+
+def entities_from_discovered_wilight(hass, api_device):
+    """Parse configuration and add WiLight fan entities."""
+    entities = []
+    for item in api_device.items:
+        if item["type"] != ITEM_FAN:
+            continue
+        index = item["index"]
+        item_name = item["name"]
+        if item["sub_type"] == FAN_V1:
+            entity = WiLightFan(api_device, index, item_name)
+        else:
+            continue
+        entities.append(entity)
+
+    return entities
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up WiLight lights from a config entry."""
+    parent = hass.data[DOMAIN][entry.entry_id]
+
+    # Handle a discovered WiLight device.
+    entities = entities_from_discovered_wilight(hass, parent.api)
+    async_add_entities(entities)
+
+
+class WiLightFan(WiLightDevice, FanEntity):
+    """Representation of a WiLights fan."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the device."""
+        WiLightDevice.__init__(self, *args, **kwargs)
+        # Initialize the WiLights fan.
+        self._direction = DIRECTION_FORWARD
```

Comment:
`DIRECTION_FORWARD` is a home assistant constant. We shouldn't mix it with device specific values.

Reply:
Done!